Build failure with GHC 9.2.1

```
Data/PrimitiveArray/Dense.hs:82:77: error:
    • Couldn't match kind ‘k’ with ‘*’
      Expected kind ‘* -> *’, but ‘v’ has kind ‘k -> *’
      ‘k’ is a rigid type variable bound by
        the data constructor ‘MDense’
        at Data/PrimitiveArray/Dense.hs:(82,1)-(83,29)
    • In the first argument of ‘VG.Mutable’, namely ‘v’
      In the type ‘(VG.Mutable v (PrimState m) e)’
      In the definition of data constructor ‘MDense’
   |
82 | data instance MutArr m (Dense v sh e) = MDense !(LimitType sh) !(VG.Mutable v (PrimState m) e)
   |                                                                             ^
cabal: Failed to build PrimitiveArray-0.10.1.1.
```

v0.10.0.0 shows the same error.

--- a/Data/PrimitiveArray/Dense.hs
+++ b/Data/PrimitiveArray/Dense.hs
@@ -79,7 +79,7 @@ instance (NFData (LimitType sh), NFData (v e)) ⇒ NFData (Dense v sh e) where
 
 
 
-data instance MutArr m (Dense v sh e) = MDense !(LimitType sh) !(VG.Mutable v (PrimState m) e)
+data instance MutArr m (Dense (v :: * -> *) sh e) = MDense !(LimitType sh) !(VG.Mutable v (PrimState m) e)
   deriving (Generic,Typeable)
 
 instance (Show (LimitType sh), Show (VG.Mutable v (PrimState m) e), VG.Mutable v (PrimState m) e ~ mv) ⇒ Show (MutArr m (Dense v sh e)) where

--- a/Data/PrimitiveArray/Sparse/BinSearch.hs
+++ b/Data/PrimitiveArray/Sparse/BinSearch.hs
@@ -63,7 +63,7 @@ type Boxed    w sh e = Sparse w  V.Vector sh e
 -- | Currently, our mutable variant of sparse matrices will keep indices and manhattan starts
 -- immutable as well.
 
-data instance MutArr m (Sparse w v sh e) = MSparse
+data instance MutArr m (Sparse w (v :: * -> *) sh e) = MSparse
   { msparseUpperBound :: !(LimitType sh)
   , msparseData       :: !(VG.Mutable v (PrimState m) e)
   , msparseIndices    :: !(w sh) -- (VG.Mutable w (PrimState m) sh)

--- a/Data/PrimitiveArray/Sparse/IntBinSearch.hs
+++ b/Data/PrimitiveArray/Sparse/IntBinSearch.hs
@@ -64,7 +64,7 @@ type Boxed    w sh e = Sparse w  V.Vector sh e
 -- | Currently, our mutable variant of sparse matrices will keep indices and manhattan starts
 -- immutable as well.
 
-data instance MutArr m (Sparse w v sh e) = MSparse
+data instance MutArr m (Sparse w (v :: * -> *) sh e) = MSparse
   { msparseUpperBound :: !(LimitType sh)
   , msparseData       :: !(VG.Mutable v (PrimState m) e)
   , msparseIndices    :: !(VU.Vector Int)

